### PR TITLE
[SPARK-27394][WEBUI] Flush LiveEntity if necessary when receiving SparkListenerExecutorMetricsUpdate (backport 2.4)

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

This PR backports #24303 to 2.4.

## How was this patch tested?

Jenkins

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala [hunk 1]
```diff
@@ -724,6 +728,31 @@ class UISeleniumSuite extends SparkFunSuite with WebBrowser with Matchers with B
     }
   }
 
+  test("Staleness of Spark UI should not last minutes or hours") {
+    withSpark(newSparkContext(
+      master = "local[2]",
+      // Set a small heart beat interval to make the test fast
+      additionalConfs = Map(
+        "spark.executor.heartbeatInterval" -> "1s",
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/ui/UISeleniumSuite.scala]: set this to 1 second due to https://issues.apache.org/jira/browse/SPARK-27419

**Comment by gatorsmile** [comment]
cc @jiangxb1987 @vanzin

**Comment by jiangxb1987** [comment]
LGTM

**Review by felixcheung — APPROVED** [review]

**Comment by vanzin** [comment]
Looks good pending tests.

**Comment by zsxwing** [comment]
Thanks! Merging to 2.4.

**Pull request closed without merging**